### PR TITLE
WT-12842 Tidy up txn upd_value's tw assignment method

### PR DESCRIPTION
**Pull request by y123456yz**
optimize txn upd_value's tw assignment method

### CODE REVIEW

**Comment by y123456yz** [comment]
[WT-12842](https://jira.mongodb.org/browse/WT-12842)

File: src/include/txn_inline.h [hunk 1]
```diff
@@ -1163,11 +1163,7 @@ __wt_txn_read_upd_list_internal(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt,
          */
         if (upd->type == WT_UPDATE_TOMBSTONE && F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE) &&
           !WT_TIME_WINDOW_HAS_STOP(&cbt->upd_value->tw)) {
-            cbt->upd_value->tw.durable_stop_ts = upd->durable_ts;
```
[the remark this answers is not in the document]
Comment (luke-pearson) [reply src/include/txn_inline.h:1166]: This seems okay to me as at most it'll try and save a tombstones data and size pointers but those will be empty. Two things can then happen, we either return the tombstone or we overwrite those entries with a standard updates data and size pointers. We won't read those values in the first case and the second is safe.

**Comment by luke-pearson** [comment]
Needs s_all.

File: src/include/txn_inline.h [hunk 2]
```diff
@@ -1163,11 +1163,7 @@ __wt_txn_read_upd_list_internal(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt,
...
-            cbt->upd_value->tw.stop_ts = upd->start_ts;
-            cbt->upd_value->tw.stop_txn = upd->txnid;
-            cbt->upd_value->tw.prepare =
-              prepare_state == WT_PREPARE_INPROGRESS || prepare_state == WT_PREPARE_LOCKED;
+            __wt_upd_value_assign(cbt->upd_value, upd);
```
Comment (korteland) [inline src/include/txn_inline.h:1166]: I dislike adding a second `WT_ACQUIRE_READ_WITH_BARRIER` for `prepare_state`. It's unavoidable since we need `__wt_upd_value_assign` to use it for correctness, but we also need to check `prepare_state` against `WT_PREPARE_RESOLVED` above. Besides not being free (performance-wise), this second read also raises the possibility of the value changing between the two reads.

It may not be a problem that the value changes between the reads, or it may not be possible. But either way it's harder to reason about than just leaving the original three lines intact.

So for that reason, I'm inclined to NAK this part of the change - I think the cleanup inside `__wt_upd_value_assign` is useful though. What do you think, @luke-pearson?

Comment (luke-pearson) [reply src/include/txn_inline.h:1166]: We do call `__wt_upd_value_assign` later in this function though. The real fix here would be to mace prepare state an argument to that function making it explicit that the caller needs to handle the ordering semantics.

**Comment by luke-pearson** [comment]
@y123456yz, @korteland and myself discussed the changes to txn_inline.h and for now we would like to drop those as it creates unintentional side effects including setting the update type and data / size pointers. We'd still like to merge the changes to the `__wt_upd_value_assign` function itself.

File: src/include/txn_inline.h (hunk 1 above)
Comment (luke-pearson) [inline src/include/txn_inline.h:1166]: Please revert this change for the reasons I mentioned in my other comment.

Comment (y123456yz) [reply src/include/txn_inline.h]: done, thanks.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Review by korteland — APPROVED** [review]
LGTM
